cni/pkg/iptables/iptables.go

```diff
@@ -236,6 +242,18 @@ func (cfg *IptablesConfigurator) appendInpodRules(hostProbeSNAT, hostProbeV6SNAT
 		"-j", ChainInpodOutput,
 	)
 
+	if !cfg.cfg.RedirectTPROXY {
```

Comment:
what's the behaviour when someone changes the mode in a live cluster?
Reading the code, i think it will only apply to new pods (i.e. we would never have both rules applied in the same pod).

Wanted to confirm that you see the same behaviour.

Reply:
Is leaving it in the old state actually a desired behavior or should we attempt to "repair" existing pods? Likely more risk in repairing but something to consider.

Reply:
We don't support pod "repair" in ambient, it's just a sidecar thing. We will need such a thing eventually IMO however, as ambient pods in theory need "repair" as much as sidecars do.

To do that, we'd need `iptables` repair logic that's safe and functional - which means we'd need something  like https://github.com/istio/istio/pull/50328 first.